content/recipes/typegoose.md

```diff
@@ -0,0 +1,346 @@
+
+This recipe will get you started working with Typegoose in NestJS.
+
+#### Getting started
+
+To begin, you'll need to install the following dependencies into your [already created NestJS app](/first-steps):
+
+```typescript
+$ npm install --save mongoose @typegoose/typegoose @m8a/nestjs-typegoose
+```
+
+Here is a short description of each package. 
+
+ - **mongoose** - the core and powerful Mongoose ODM library
+ - **@typegoose/typegoose** - the Typegoose library
+ - **@m8a/nestjs-typegoose** - the package containing the Typegoose module for plugging in Typegoose into Nest
+
+> info **Hint** Some of the content in this recipe was taken from the [documentation of the Typegoose module](https://nestjs-typegoose.m8a.io/). You can get further details there and also further details about Typegoose at their [docs website](https://typegoose.github.io/typegoose/docs/guides/quick-start-guide). 
+
+#### Setting up the DB Connection
+For the next step, we will need to configure the connection to the MongoDB database. To do that, we'll use the `TypegooseModule.forRoot` static method. 
+
+```typescript
+@@filename(app.module)
+import { Module } from "@nestjs/common";
+import { TypegooseModule } from "@m8a/nestjs-typegoose";
+
+@Module({
+  imports: [
+    TypegooseModule.forRoot("mongodb://localhost:27017/otherdb", {
+      // other connection options
+    }),
+    CatsModule
+  ]
+})
+export class ApplicationModule {}
+@@switch
+import { Module } from "@nestjs/common";
+import { TypegooseModule } from "@m8a/nestjs-typegoose";
+
+@Module({
+  imports: [
+    TypegooseModule.forRoot("mongodb://localhost:27017/otherdb", {
+      // other connection options
+    }),
+    CatsModule
+  ]
+})
+export class ApplicationModule {}
+```
+The second parameter of the `TypegooseModule.forRoot` static method entails [the additional connection options](https://mongoosejs.com/docs/connections.html#options) available from Mongoose.
+
+Also notice we imported the `CatsModule`. We will create that module shortly.
+
+If you have requirements to use multiple databases, you can also implement [multiple connections to different databases](https://nestjs-typegoose.m8a.io/docs/multiple-connections).
+
+#### Creating Entities
+
+Now that you have the dependencies installed and the database connection is set up, let's create our first entity. This is a very simplified example.
+
+```typescript
+@@filename(cat.entity)
+import { Prop } from "@typegoose/typegoose";
+
+export class Cat {
+  @Prop({ required: true })
+  public name!: string;
+}
+@@switch
+import { Prop } from "@typegoose/typegoose";
+
+export class Cat {
+  @Prop({ required: true })
+  public name!: string;
```

Comment:
since this will be the JS version, I guess we should do
```suggestion
  name
```

Reply:
the same applies to others snippets below

Reply:
@micalevisk - None of this will work with JavaScript, I just didn't know how to get the filename header/ tab without the switcher feature in it. So, I just doubled up the code.  

Scott

Reply:
oh, right. So we could just drop the `@@filename(app.module)` and so the JS version

Reply:
Done. I added the file names as a code comment.

Scott